File: C++/I/08-LINEEDITS/1/colorlineedit.cpp
#include "colorlineedit.h"
#include <QApplication>
#include <QKeyEvent>
#include <iostream>

ColorLineEdit::ColorLineEdit(QWidget *parent):
   QLineEdit(parent){}

void ColorLineEdit::focusInEvent(QFocusEvent *)
{
    setStyleSheet("background-color: darkGreen; color: white");
}

void ColorLineEdit::focusOutEvent(QFocusEvent *)
{
    setStyleSheet("background-color: Window; color: WindowText");
}

File: Python/I/08-LINEEDITS/1/colorlineedit.py
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *

class ColorLineEdit(QLineEdit):
    def focusInEvent(self, event):
        self.setStyleSheet('background-color: darkGreen; color: white')
        self.selectAll()
        QApplication.sendEvent(self,
            QKeyEvent(QEvent.KeyPress, Qt.Key_Right, Qt.NoModifier))
        self.deselect()

    def focusOutEvent(self, event):
        self.selectAll()
        self.deselect()
        self.setStyleSheet('background-color: Window; color: WindowText')
